### PR TITLE
develop

### PR DESCRIPTION
**Pull request by savchuksavchuk**

### CODE REVIEW

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,64 @@
...
+      if (order === 'desc') {
+        return a[sortBy] <= b[sortBy] ? 1 : -1;
+      }
+
+      return a[sortBy] <= b[sortBy] ? -1 : 1;
+    }
+
+    if (sortBy === 'grades') {
+      const first: number = a[sortBy]
+        .reduce((x: number, y: number): number => x + y) / a[sortBy].length;
```
Comment (Bohdan-Maksymiuk) [inline src/sortStudents.ts:45]: pass this logic into separate function and reuse it

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,64 @@
...
+    if (sortBy === 'married') {
+      if (order === 'desc') {
+        return a[sortBy] <= b[sortBy] ? 1 : -1;
+      }
+
+      return a[sortBy] <= b[sortBy] ? -1 : 1;
+    }
+
+    if (sortBy === 'grades') {
+      const first: number = a[sortBy]
```
Comment (Bohdan-Maksymiuk) [inline src/sortStudents.ts:44]: use switch for such predefined cases

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,16 +1,64 @@
...
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const result: Student[] = [...students];
+
+  result.sort((a: Student, b: Student): number => {
```
Comment (Bohdan-Maksymiuk) [inline src/sortStudents.ts:26]:
```suggestion
  return [...students].sort((a: Student, b: Student): number => {
```
you may just return it

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]

**Review by liliia-livandovskaya — CHANGES_REQUESTED** [review]
GJ! 👍  Pay attention to comments

**Review by polosanya — APPROVED** [review]
Well done!

**Pull request closed without merging**